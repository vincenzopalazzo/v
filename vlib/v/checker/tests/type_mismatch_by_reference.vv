import math

struct Vector{
	x f64
	y f64
}

struct Vector3d{
	x f64
	y f64
	z f64
}

struct Position{
	x f64
	y f64
}

struct Position3d{
	x f64
	y f64
	z f64
}

fn new_position (v &Vector, p &Position, time f32) Vector{
	x := v.x*time
	y := v.y*time
	return Vector{p.x + x, p.y + y}
}

fn acceleration(u &Vector, v &Vector, time f32) Vector{
	x := u.x+v.x/time
	y := u.y+u.y/time
	return Vector{x, y}
}

fn magnitude3d(v3d &Vector3d) f64{
	return math.sqrt(math.pow(v3d.x,2)+math.pow(v3d.y,2)+math.pow(v3d.z,2))
}

fn magnitude(v &Vector) f64{
	return math.sqrt(math.pow(v.x, 2)+math.pow(v.y, 2))

}

fn unit_vector(v &Vector) Vector{
	x := v.x/magnitude(v)
	y := v.y/magnitude(v)
	return Vector{x, y}
}

fn unit_vector3d(v &Vector3d) Vector3d{
	x := v.x/magnitude3d(v)
	y := v.y/magnitude3d(v)
	z := v.z/magnitude3d(v)
	return Vector3d{x, y, z}
}

mut object := Vector{4,3}

println(acceleration(Vector{0,0}, object, 5))

