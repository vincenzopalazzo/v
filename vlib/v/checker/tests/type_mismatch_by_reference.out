vlib/v/tests/known_errors/testdata/c_err_with_def.vv:61:22: error: cannot use `Vector` as `&Vector` in argument 1 to `acceleration`
   59 | mut object := Vector{4,3}
   60 |
   61 | println(acceleration(Vector{0,0}, object, 5))
